### ISSUE TITLE
Can't get OTR to work

### ISSUE THREAD

**Issue by updavi** [profanity-im/profanity#1614]
After upgrading Profanity as part of the upgrade from OpenBSD 6.9 tp 7.0, OTR chat has stopped working. It's no longer possible to start an encrypted session, either manually with `/otr start` or automatically with `/otr policy always`.

Reviewing the `profanity.log` file, I see the following error when I try to start an OTR session:
```
WRN: Invalid carbon received, from: my@contact.name/XMPP
```
This happens with contacts who I previously chatted over OTR with, using Profanity 0.10.0.

This happens for all my XMPP accounts in Profanity, not just one,

## Expected Behavior

OTR session starts

## Current Behavior

Nothing happens in the UI, no OTR session starts. Message about "invalid carbon" is logged to profanity.log

## Possible Solution

No obvious work arounds

## Steps to Reproduce (for bugs)

1. On OpenBSD, `pkg_add profanity` and configure the client
2. Attempt to start an OTR session someone whose client supports OTR (and ideally where you previously commuicated with them using OTR)

## Context

OTR encrypted communication with Profanity is no longer possible

## Environment
* Libotr 4.1.1
* Profanity 0.11.0
* XMPP library libstrophe
* OpenBSD 7.0

**Comment by jubalh** [comment]
I doubt there changed anything regarding the release.

OTR is just brittle and it's hard for people to use, which is why so many clients removed OTR support.

I still like and use OTR. It works well for me.

I'm not entirely sure what breaks OTR. But basically it's designed for a one client to one client world. Multiclient support was not intended.

So please test what happens if you are online with only one client, and your partner too. Does it work then?

It could be that one party goes online with another client, maybe with a higher prio, and then we try to send to this client, breaking OTR.

See setting `/resource`.

**Comment by jubalh** [comment]
BTW https://github.com/profanity-im/profanity/releases/tag/0.11.1 is the latest release.

**Comment by updavi** [comment]
> BTW https://github.com/profanity-im/profanity/releases/tag/0.11.1 is the latest release.

I'm using OpenBSD 7, which packages 0.11.0

My conversation partners are all using one client only, and I am also only using one client.

I can't downgrade to 0.10.0 because that is not available on my system.

`/resource` shows that all display options are turned on, but I don't see anything to give me more information.

**Comment by jubalh** [comment]
> I'm using OpenBSD 7, which packages 0.11.0

They should update. It's a point release.

> I can't downgrade to 0.10.0 because that is not available on my system.

If you can build from source you could try it.

**Comment by pukkamustard** [comment]
I am experiencing the same issue (OTR fails to start).

I'm running prosody 0.11.1 on a Linux system (Guix):

```
$ profanity --version
Profanity, version 0.11.1
Copyright (C) 2012 - 2019 James Booth <boothj5web@gmail.com>.
Copyright (C) 2019 - 2021 Michael Vetter <jubalh@iodoru.org>.
License GPLv3+: GNU GPL version 3 or later <https://www.gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Build information:
XMPP library: libmesode
Desktop notification support: Enabled
OTR support: Enabled (libotr 4.1.1)
PGP support: Enabled (libgpgme 1.15.1)
OMEMO support: Enabled
C plugins: Enabled
Python plugins: Enabled (3.8.2)
GTK icons/clipboard: Enabled
```

I get the same messages in the logs as mentioned above (`WRN: Invalid carbon received, from: _____`). I've tried this with same version of profanity on both sides.

Maybe something did change from 0.10.0 to 0.11.0?

**Comment by pukkamustard** [comment]
> Maybe something did change from 0.10.0 to 0.11.0?

I downgraded to 0.10.0 and still am experiencing this issue. So not something specific to 0.11.x maybe...

**Comment by jubalh** [comment]
> I downgraded to 0.10.0 and still am experiencing this issue

Tha'ts what I thought ;)

**Comment by pukkamustard** [comment]
I managed to get OTR working with a local development server and two profanity instances connected (with version 0.11.1). So really does not seem to be something in general wrong with OTR in 0.11.1. I think it has something to do with Message Carbons.

I tested following on my main server (the one where OTR is currently not working):

1. User A starts an OTR session to user B with `/otr start`
2. On the XML Console of B the message appears with the OTR query (".?OTR?v2? This user has requested an Off-the-Record private conversation. However, ...")
3. Message does not appear in chat window of user B neither is an OTR response sent.

To test I manually send an OTR query:

1. User A sends an unencrypted message with content "?OTR?v2?" to user B (just by entering this into the chat).
2. On the XML console of B the message appears. It does not appear in the chat window and B does not send anything in response.

Now if I set carbons off with `/carbons off` and do this:

1. User A sends an unencrypted message with content "?OTR?v2?" to user B (just by entering this into the chat).
2. At user B: the message appears in the XML Console and an OTR response is sent with (body: " ?OTR:AAIC..."). Nothing appears in the chat window.
3. User A does not respond to the OTR message from B (expected)

My local development server does not have message carbons enabled. I suspect that is the reason why I can start OTR there.

Hypothesis:

If there is a "<private xmlns="urn:xmpp:carbons:2"/>" element in the OTR query (message with "?OTR?v2?" string) then profanity does not correctly handle it. This might also explain the warning in the log file ("Invalid carbon received, from ....").
